workspace.spec

```diff
@@ -572,6 +572,378 @@ module Workspace {
 		string description;
 	} ProvenanceAction;
 
+	/* Credit Metadata-related objects and fields
+
+		Workspace objects have a list of zero or more sets of credit metadata (CM), or citation information for the object. This information allows data imported into KBase to retain the appropriate details for users wishing to cite the data source, and for those who contributed to its creation to be credited for their work.
+
+		CM is stored in a list as part of the object provenance, but unlike other provenance information, which is immutable, CM can be updated.
+
+		Credit metadata is added after object creation, and requires the user to be a workspace admin. Username, timestamp, and credit metadata schema version are stored along with the CM, which allows tracking of CM changes.
+
+		To update existing CM, a new set of credit metadata is added to the list.
+
+		When requesting credit metadata, the last (i.e. most recent) entry in the list is returned.
+
+		In the following entries, "resource" is used to refer to the workspace object that the CM pertains to.
+	*/
+
+	/* EventDate
+
+		Represents an event in the lifecycle of a resource and the date it occurred on.
+
+		See https://support.datacite.org/docs/datacite-metadata-schema-v44-recommended-and-optional-properties#8-date for more information on the events (below).
+
+		Both event and date are required fields.
+
+		string date - the date associated with the event. The date may be in the format
+			YYYY, YYYY-MM, or YYYY-MM-DD.
+		string event - the event that occurred; events must be from the following list:
+			"accepted"
+			"available"
+			"copyrighted"
+			"collected"
+			"created"
+			"issued"
+			"submitted"
+			"updated"
+			"valid"
+			"withdrawn"
+			"other"
+	*/
+	typedef structure {
+		string date;
+		string event;
+	} EventDate;
+
+	/* FundingReference
+
+		Represents a funding source for a resource, including the funding body and the grant awarded.
+
+		The 'funder_name' field is required; all others are optional.
+
+		Recommended resources for organization identifiers include:
+			- Research Organization Registry, http://ror.org
+			- International Standard Name Identifier, https://isni.org
+			- Crossref Funder Registry, https://www.crossref.org/services/funder-registry/
+
+		Some organizations may have a digital object identifier (DOI).
+
+		string funder_id (optional) - persistent unique identifier for the funder in the format
+			<database name>:<identifier within database>
```

Comment:
examples for the funder_id and funder_name fields would be helpful

Reply:
I've gone through and added examples for as many fields as possible

Reply:
:+1: